efabless/openlane2
# Fix missing antenna properties in generated LEF views

* `Magic.WriteLef`:
	* Fixed missing `ANTENNANGATEAREA` and `ANTENNADIFFAREA` in pins of generated LEF view. This is done by using GDSII views for PDK cells and user-defined macros instead of LEF views.

* `Odb.*`:
	* (Internal) changed `reader.cells` to hold cell *objects* instead of *names*

* Created `Odb.CheckMacroAntennaProperties`:
	* Checks user-defined macros for pin antenna properties in LEF views. For pins not marked as `POWER`, `GROUND` or `ANALOG`:
	 * Checks input pins for `ANTENNAGATEAREA`.
	 * Checks output pins for `ANTENNADIFFAREA`.
	 * Checks inout pins for both `ANTENNAGATEAREA` and `ANTENNADIFFAREA`.
	 * Describes said pins per macro in `report.yaml`.

* `Odb.CheckDesignAntennaProperties`:
	* Same as `Odb.CheckMacroAntennaProperties` but checks the generated design LEF view.

* `Classic`:
  * Added `Odb.CheckMacroAntennaProperties`.
  * Added `Odb.CheckDesignAntennaProperties`.

# Tool Updates
* Updated OpenROAD to `0889970`
	* Patched in method `MTerm.hasDiffArea` to check if a macro's pin has `ANTENNADIFFAREA`.
	* Patched in method `TechAntennaPinModel.hasGateArea` callable via return value of `MTerm.getAntennaModel` to check if a macro's pin has `ANTENNAGATEAREA`.
* Ignored run directories in `.gitignore`

# CI
* Fixed wrong cache restore path for gf180mcu

# Testing

* Remove `011-bad-lef` in `magic.writelef` as LEF file are no longer read by `Magic.WriteLef`. 
* Add `011-sucess-antenna-gate-area` for `magic.writelef`
* Add `012-sucess-antenna-diff-area` for `magic.writelef`
* Add unit tests for `Odb.CheckDesignAntennaProperties` and `Odb.CheckMacroAntennaProperties`

---

Depends on https://github.com/efabless/openlane2-step-unit-tests/pull/25
Depends on https://github.com/efabless/openlane2-ci-designs/pull/20
